lib/stimulus_reflex/element.rb

```diff
@@ -4,10 +4,11 @@ class StimulusReflex::Element < OpenStruct
   attr_reader :attributes, :data_attributes
 
   def initialize(attrs = {})
-    @attributes = HashWithIndifferentAccess.new(attrs || {})
-    @data_attributes = attributes.each_with_object(HashWithIndifferentAccess.new) { |(key, value), memo|
-      memo[key.delete_prefix("data-")] = value if key.start_with?("data-")
-    }
+    @attributes = HashWithIndifferentAccess.new(attrs.to_h["attrs"] || {})
+    @data_attributes = (attrs.to_h["dataset"] || {}).each_with_object(HashWithIndifferentAccess.new) { |(key, value), memo|
+      memo[key.delete_prefix("data-")] = value
+    }.freeze
+
```

Comment:
Not sure the `to_h` call is necessary on `attrs`. Did you hit conditions where the argument wasn't a `Hash`?

Reply:
Nope I didn't face a condition where it wasn't a Hash. It's just to be extra safe because it's now a two level deep Hash and before it was just one level since I now pass the whole `data` object in there.

But maybe we should rename that local variable in there from `attrs` to `data` since it's not just the `attrs` anymore.